spark-cassandra-connector/src/main/scala/org/apache/spark/metrics/CassandraConnectorSource.scala

```diff
@@ -58,16 +58,41 @@ object CassandraConnectorSource {
 
   /** This method sets the given instance of [[CassandraConnectorSource]] as the current active instance.
     * It is allowed to call this method only once for the same `SparkEnv` instance which is currently
-    * up and running.
+    * up and running or this is the executor environment.
     */
   private def maybeSetInstance(ccs: CassandraConnectorSource): Unit = synchronized {
     val curEnv = SparkEnv.get
+    val (_instance, _env) = _state
     assert(ccs != null, "ccs cannot be null")
```

Comment:
I really meant assertion. This is not intended to be used internally only and valid use of it should never cause those cases to occur.